shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/config/datasource/pool/destroyer/DataSourcePoolDestroyerFactoryTest.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.shardingsphere.infra.config.datasource.pool.destroyer;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+
+import java.sql.SQLException;
+
+import com.zaxxer.hikari.HikariDataSource;
+
+import org.apache.commons.dbcp2.BasicDataSource;
+import org.junit.Test;
+
+public final class DataSourcePoolDestroyerFactoryTest {
+    
+    @Test(timeout = 60000)
+    public void assertDestroyForHikari() throws InterruptedException {
+        try {
+            HikariDataSource dataSource = new HikariDataSource();
+            DataSourcePoolDestroyerFactory.destroy(dataSource);
+            while (!dataSource.isClosed()) {
+                // Test will fail by timeout if dataSource is not closed.
+                Thread.sleep(10L);
+            }
+        } catch (SQLException ex) {
+            fail();
+        }
+    }
+
+    @Test
+    public void assertDestroyForDefault() throws InterruptedException {
+        try {
+            BasicDataSource dataSource = new BasicDataSource();
+            DataSourcePoolDestroyerFactory.destroy(dataSource);
+            assertThat(dataSource.isClosed(), is(true));
+        } catch (SQLException ex) {
+            fail();
+        }
```

Comment:
Is it necessary to catch exception and assert fail here, how about throw SQLException in method signature?